internal/controller/controllers/agent_controller.go

```diff
@@ -204,6 +209,95 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	return r.updateStatus(ctx, log, agent, &h.Host, h.ClusterID, nil, false)
 }
 
+func (r *AgentReconciler) shouldApproveMoreCSRs(node *corev1.Node) bool {
+	return !isNodeReady(node)
+}
+
+// Validate that the CSR can be approved
+func (r *AgentReconciler) shouldApproveCSR(csr *certificatesv1.CertificateSigningRequest, agent *aiv1beta1.Agent, validateNodeCsr nodeCsrValidator) (bool, error) {
+	x509CSR, err := getX509ParsedRequest(csr)
+	if err != nil {
+		return false, err
+	}
+	if !isCsrAssociatedWithAgent(x509CSR, agent) {
+		return false, nil
+	}
+
+	if !isCsrInTimeSpan(agent, csr) {
```

Comment:
Let's ask Joel why the timeSpan check is relevant and document it. The motivation may be security-related and we should document it :D 

Otherwise, we can probably just remove this check

Reply:
He says its purpose is to avoid approving out of date CSRs. https://coreos.slack.com/archives/CUPJTHQ5P/p1644320238911699?thread_ts=1643629029.082179&cid=CUPJTHQ5P
Can be removed

Reply:
Removed